server/src/main/java/org/candlepin/dto/rules/v1/PoolTranslator.java

```diff
@@ -80,7 +80,8 @@ public PoolDTO populate(ModelTranslator modelTranslator, Pool source, PoolDTO de
         // Process nested objects if we have a model translator to use to the translation...
         if (modelTranslator != null) {
 
-            Set<Product> products = source.getProvidedProducts();
+            Collection<Product> products = source.getProduct() == null ? null :
```

Comment:
These need to be n-tier aware translations

Reply:
@sonalidhome @Ceiu 
Do we need to combined stacked entitlements provided products here like we are doing in other pool translators?

Reply:
Based on the brief discussion I posed the other day during post-scrum, these particular bits where we copy the products to the pool DTO should be recursive and grab provided products of the provided products.

Reply:
Updated the code to support n-tier.